changelog/fragments/relax-csv-name-validation.yaml

```diff
@@ -0,0 +1,8 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Resolved an issue that caused bundle validation to unnecessarily restrict CSV names
```

Comment:
I don't think this is enough information to be helpful to a user. Even linking to https://github.com/operator-framework/api/pull/63 is pretty opaque.

Reply:
Done.